mlflow/server/job_api.py

```diff
@@ -0,0 +1,91 @@
+"""
+Internal job APIs for UI invocation
+"""
+
+import json
+import os
+from typing import Any
+
+from fastapi import APIRouter, HTTPException
+from fastapi import status as http_status
+from pydantic import BaseModel
+
+from mlflow.entities._job import Job as JobEntity
+
+job_api_router = APIRouter(prefix="/ajax-api/3.0/jobs", tags=["Job"])
+
+
+_JOB_FUNCTION_ALLOW_LIST = [
+    # Add built-in job functions here
+]
+
+if extra_fun_allow_list_env := os.environ.get("_MLFLOW_JOB_FUNCTION_EXTRA_ALLOW_LIST"):
+    # This is for injecting allowlisted job functions in tests
+    _JOB_FUNCTION_ALLOW_LIST += extra_fun_allow_list_env.split(",")
+
+
+class Job(BaseModel):
+    """
+    Pydantic model for job query response.
+    """
+
+    job_id: str
+    creation_time: int
+    function_fullname: str
+    params: dict[str, Any]
+    timeout: float | None
+    status: str
+    result: Any
+    retry_count: int
+
+    @staticmethod
+    def from_job_entity(job: JobEntity):
+        return Job(
+            job_id=job.job_id,
+            creation_time=job.creation_time,
+            function_fullname=job.function_fullname,
+            params=json.loads(job.params),
+            timeout=job.timeout,
+            status=str(job.status),
+            result=job.parsed_result,
+            retry_count=job.retry_count,
+        )
+
+
+@job_api_router.get("/{job_id}", response_model=Job)
+def query_job(job_id: str) -> Job:
+    from mlflow.server.jobs import query_job
+
+    job = query_job(job_id)
+    return Job.from_job_entity(job)
+
+
+class SubmitJobPayload(BaseModel):
+    function_fullname: str
```

Comment:
How are we going to support this API on the managed MLflow? Will we have a completely separate API with an equivalent feature?

Reply:
managed MLflow send requests to backend through pubic REST endpoint, so that they should have specific endpoint definition for each type job. 

For OSS it is unneccessary, it causes the proto code exploding .

In front-end code, we can encapsulation the 2 backends endpoint invocation by universal JS helper functions easily.
e.g. we can pre-define typescript function:
```
function submit_evaluation_job(...):
   ....
```
and For OSS and Databricks front-end code, the `submit_evaluation_job` typescript function can have different implementation.